src/deroff.rs

```diff
@@ -1168,6 +1155,308 @@ fn deroff_files(files: &[String]) -> io::Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_text_arg() {
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Hello World!".into();
+    assert!(deroffer.text_arg());
+    assert_eq!(deroffer.s, " World!");
+    assert_eq!(deroffer.output.take(), "Hello");
+
+    let mut deroffer = Deroffer::new();
+    assert!(!deroffer.text_arg());
+    assert!(deroffer.s.is_empty());
+    assert!(deroffer.output.take().is_empty());
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "\t\n\t           \t\n".into();
+    assert!(!deroffer.text_arg());
+    assert_eq!(deroffer.s, "\t\n\t           \t\n");
+    assert!(deroffer.output.take().is_empty());
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = r"\r".into();
+    assert!(!deroffer.text_arg());
+    assert!(deroffer.s.is_empty());
+    assert_eq!(deroffer.output.take(), "r");
+
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Applebees".into();
+    assert!(deroffer.text_arg());
+    assert_eq!(deroffer.s, "");
+    assert_eq!(deroffer.output.take(), "Applebees");
+}
+
+#[test]
+fn test_font() {
+    let mut deroffer = Deroffer::new();
+    deroffer.s = r"\f(aa)lemon".into();
+    assert!(deroffer.font());
+    assert_eq!(deroffer.s, ")lemon");
+    assert!(!deroffer.font());
+    assert_eq!(deroffer.s, ")lemon");
+}
+
+#[test]
+fn test_numreq() {
+    let mut deroffer = Deroffer::new();
+    deroffer.s = "Hello World!".into();
+    assert!(!deroffer.numreq());
+    assert_eq!(deroffer.s, "Hello World!");
+    assert!(deroffer.output.take().is_empty());
+
+    deroffer.s = r"\w'Apple'".into();
+    assert!(deroffer.numreq());
+    assert!(deroffer.s.is_empty());
+    assert!(deroffer.output.take().is_empty());
+
+    deroffer.s = r"\w'Hello\tWorld!'".into();
+    assert!(deroffer.numreq());
+    assert_eq!(deroffer.s, "!'");
+    assert!(deroffer.output.take().is_empty());
```

Comment:
Different from python behavior.
```python
>>> d = Deroffer()
>>> d.s = "\w'Hello\tWorld!'"
>>> d.numreq()
True
>>> d.s
"\tWorld!'"
>>> d.output
[]
```

Reply:
@charliethomson look at these tonight

Reply:
@pickfire I was testing these while I was going, I'm not sure why but it seems to matter here whether or not you use a raw string (`r"whatever"`) or a standard string (`"whatever"`):
```py
>>> d.s = r"\w'Hello\tWorld!'"
>>> d.numreq()
True
>>> d.s
"!'"
>>> d.output
[]
>>> d.s = "\w'Hello\tWorld!'"
>>> d.numreq()
True
>>> d.s
"\tWorld!'"
>>> d.output
[]
```

Reply:
Oh, I think we may need to use raw strings, raw strings interpret `\` as escapes IIRC.